packages/notifi-frontend-client/lib/NotifiFrontendClient.ts

```diff
@@ -73,4 +81,61 @@ export class NotifiFrontendClient {
 
     await Promise.all([saveAuthorizationPromise, saveRolesPromise]);
   }
+
+  async beginLoginViaTransaction({
+    walletBlockchain,
+    walletAddress,
+  }: BeginLoginProps): Promise<Types.BeginLogInByTransactionResult> {
+    const { dappAddress } = this._configuration;
+
+    const result = await this._service.beginLogInByTransaction({
+      walletAddress: walletAddress,
+      walletBlockchain: walletBlockchain,
+      dappAddress,
+    });
+
+    const nonce = result.beginLogInByTransaction.nonce;
+
+    if (nonce !== null) {
+      const ruuid = crypto.randomUUID();
+      this._clientRandomUuid = ruuid;
+      const encoder = new TextEncoder();
+      const data = encoder.encode(nonce + ruuid);
+
+      const hashBuffer = await crypto.subtle.digest('SHA-256', data);
+
+      const hashArray = Array.from(new Uint8Array(hashBuffer));
+      const logValue =
+        'Notifi Auth: 0x' +
+        hashArray.map((b) => b.toString(16).padStart(2, '0')).join('');
+      return { nonce: logValue };
+    } else throw new Error('Failed to begin login process');
```

Comment:
nit: prefer early return pattern to avoid deep indentation 

```
if (nonce === null) {
  throw ...
}

...
```

Reply:
good call - so the code doesn't have to find it at the end

Reply:
> Nice, looks good to me.
> 
> Stupid question, but `npx lerna run build` completes with no errors, right? We should probably put that on a pre-commit hook somewhere.

<img width="1223" alt="Screen Shot 2022-08-18 at 11 16 39 AM" src="https://user-images.githubusercontent.com/108922885/185465812-e384041e-739c-4e25-a7a7-bb039d85421c.png">

Yes, it does! Let me look into adding the command in a pre-commit and how that would work.